metaseq/scripts/generation_benchmarks.py

```diff
@@ -0,0 +1,279 @@
+        "https://dl.fbaipublicfiles.com/opt/v1_20220502/2.7b/reshard-model_part-3.pt",
+    ]
+    links_to_data["6.7b"] = [
+        "https://dl.fbaipublicfiles.com/opt/v1_20220502/6.7b/reshard-model_part-0.pt",
+        "https://dl.fbaipublicfiles.com/opt/v1_20220502/6.7b/reshard-model_part-1.pt",
+    ]
+    links_to_data["13b"] = [
+        "https://dl.fbaipublicfiles.com/opt/v1_20220502/13b/reshard-model_part-0.pt",
+        "https://dl.fbaipublicfiles.com/opt/v1_20220502/13b/reshard-model_part-1.pt",
+    ]
+    links_to_data["30b"] = [
+        "https://dl.fbaipublicfiles.com/opt/v1_20220502/30b/reshard-model_part-0.pt",
+        "https://dl.fbaipublicfiles.com/opt/v1_20220502/30b/reshard-model_part-1.pt",
+    ]
+    links_to_data["66b"] = [
+        "https://dl.fbaipublicfiles.com/opt/v1_20220502/66b/reshard-model_part-0-shard0.pt",
+        "https://dl.fbaipublicfiles.com/opt/v1_20220502/66b/reshard-model_part-1-shard0.pt",
+        "https://dl.fbaipublicfiles.com/opt/v1_20220502/66b/reshard-model_part-2-shard0.pt",
+        "https://dl.fbaipublicfiles.com/opt/v1_20220502/66b/reshard-model_part-3-shard0.pt",
+        "https://dl.fbaipublicfiles.com/opt/v1_20220502/66b/reshard-model_part-4-shard0.pt",
+        "https://dl.fbaipublicfiles.com/opt/v1_20220502/66b/reshard-model_part-5-shard0.pt",
+        "https://dl.fbaipublicfiles.com/opt/v1_20220502/66b/reshard-model_part-6-shard0.pt",
+        "https://dl.fbaipublicfiles.com/opt/v1_20220502/66b/reshard-model_part-7-shard0.pt",
+    ]
+
+    if key not in links_to_data.keys():
+        raise AssertionError(
+            "You passed model name "
+            + str(key)
+            + ". Please make sure that model name is in "
+            + str(list(links_to_data.keys()))
+        )
+
+    links_to_data["dependencies"] = [
+        "http://dl.fbaipublicfiles.com/metaseq_benchmark_dependencies.tar.gz"
+    ]
+
+    # download and unzip dependencies
+    for dependency in links_to_data["dependencies"]:
+        urllib.request.urlretrieve(dependency, "./dependencies.tar.gz")
+        file = tarfile.open("./dependencies.tar.gz")
+        file.extractall()
+
+    # download model checkpoint
+    for shard in links_to_data[key]:
+        file_name = shard.split("/")[-1]
+        urllib.request.urlretrieve(shard, "./dependencies/" + file_name)
+
+
+def download_prompts():
+    link_to_prompt = "https://dl.fbaipublicfiles.com/CommonCrawlSmall.jsonl"
+    urllib.request.urlretrieve(link_to_prompt, "./CommonCrawlSmall.jsonl")
+    pass
+
+
+def cleanup_data():
+    shutil.rmtree("./dependencies")
+    os.remove("./dependencies.tar.gz")
+
+
+def cleanup_prompts(prompt_path):
+    os.remove(prompt_path)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--model_name", nargs="+")
+    parser.add_argument("--prompt_path", nargs="?")
+    parser.add_argument("--padding_size", nargs="?", default="512")
```

Comment:
can padding be automatically computed from the inputs?

Reply:
Yes as in padding = length of longest sentence. But in the same spirit as having more sentences, having a higher padding size would mean more generated words and hence a more true average?

Reply:
In practice, our batching tries to group similarly lengthed examples together to minimize padding. So some padding is realistic, but we actively try to minimize it.

Reply:
hmm I see. So if I have one sentence (as per comment above) and have that run, say, 1000 times, will the padding being equal to the sentence length make sense? In which case padding will no longer be an argument